src/telemetry/Manager.h

```diff
@@ -50,6 +58,101 @@ class Manager
 	 */
 	virtual void InitPostScript();
 
+	/**
+	 * Instances returned by CollectMetrics().
+	 */
+	enum MetricType
+		{
+		DOUBLE_COUNTER,
+		INT_COUNTER,
+		DOUBLE_GAUGE,
+		INT_GAUGE,
+		DOUBLE_HISTOGRAM,
+		INT_HISTOGRAM,
+		};
+
+	struct CollectedMetric
```

Comment:
The twist is that `CollectedMetric` currently does hold histograms, they just don't have a value. The motivation was that script land can list metrics regardless of their type with a single call.

However, given we have a separate `collect_histogram_metrics` already and that we even create a separate log, maybe that was too ambitious. We can ask them to invoke `collect_metrics()` and `collect_histogram_metrics()` instead and have things simpler here, too.

(@rsmmr - ack / nack?)

Reply:
@Neverlord - I've worked on your suggestions and switched to std::variant for both the `CollectedValueMetric` and the `HistogramMetric` (including adding explicit buckets).

I think it looks better, but there's are some awkward `std::holds_alternative` in the `AsMetricRecord` and `AsHistogramMetricRecord` render functions. This is mostly because a separate ambition to make the `int` values available to script land if we have them.

Please take another look. I'm going to resolve the other comments that went into the same general direction. Thanks!